hawtio-util/src/test/java/io/hawt/util/FilesTest.java

```diff
@@ -0,0 +1,298 @@
+    // Method is not expected to return due to exception thrown
+  }
+
+  @Test
+  public void copyIOException() throws Exception, IOException {
+    // Setup mocks
+    final File source = PowerMockito.mock(File.class);
+    final File target = PowerMockito.mock(File.class);
+    final File targetParent = PowerMockito.mock(File.class);
+
+    // Arrange
+    PowerMockito.when(source.exists()).thenReturn(true);
+    PowerMockito.when(target.exists()).thenReturn(false);
+
+    PowerMockito.when(targetParent.getAbsolutePath()).thenReturn("/tmp/targetParent");
+    PowerMockito.when(targetParent.exists()).thenReturn(false);
+    PowerMockito.when(targetParent.mkdirs()).thenReturn(false);
+    PowerMockito.when(target.getParentFile()).thenReturn(targetParent);
+
+    // Act
+    thrown.expect(IOException.class);
+    Files.copy(source, target);
+    // Method is not expected to return due to exception thrown
+  }
+
+  @PrepareForTest({IOHelper.class, Files.class})
+  @Test
+  public void copy() throws Exception, IOException {
+    // Setup mocks
+    final File source = PowerMockito.mock(File.class);
+    final File target = PowerMockito.mock(File.class);
+    final File targetParent = PowerMockito.mock(File.class);
+    final FileInputStream is = PowerMockito.mock(FileInputStream.class);
+    final FileOutputStream os = PowerMockito.mock(FileOutputStream.class);
+    PowerMockito.mockStatic(IOHelper.class);
+
+    // Arrange & Act
+    PowerMockito.when(source.exists()).thenReturn(true);
+    PowerMockito.when(target.getParentFile()).thenReturn(targetParent);
+    PowerMockito.whenNew(FileInputStream.class).withParameterTypes(File.class).withArguments(isA(File.class)).thenReturn(is);
+    PowerMockito.whenNew(FileOutputStream.class).withParameterTypes(File.class).withArguments(isA(File.class)).thenReturn(os);
+
+    PowerMockito.when(target.exists()).thenReturn(true);
+    PowerMockito.when(targetParent.exists()).thenReturn(false);
+    PowerMockito.when(targetParent.mkdirs()).thenReturn(false);
+    Files.copy(source, target);
+    // Method returns void, testing that no exception is thrown
+
+    PowerMockito.when(target.exists()).thenReturn(false);
+    PowerMockito.when(targetParent.mkdirs()).thenReturn(true);
+    Files.copy(source, target);
+    // Method returns void, testing that no exception is thrown
+    
+    PowerMockito.when(targetParent.exists()).thenReturn(true);
+    Files.copy(source, target);
+    // Method returns void, testing that no exception is thrown
+  }
+
+  @Test
+  public void recursiveDelete() throws Exception {
+    // Setup mocks
+    final File file = PowerMockito.mock(File.class);
+    final File file1 = PowerMockito.mock(File.class);
+    final File file21 = PowerMockito.mock(File.class);
+    final File file22 = PowerMockito.mock(File.class);
+    final File file2 = PowerMockito.mock(File.class);
+    final File file3 = PowerMockito.mock(File.class);
+
+    // Arrange, Act & Assert result
```

Comment:
This is a pretty weird comment. Was this auto-generated?

Reply:
T clarify: you may as well just remove it if this comment tells us that "i'm about to do everything in one block"

Reply:
Alternatively replace comment with `// Everything`

Reply:
Fine, i'll remove the comments.